docs/changelog/4_x.rst

```diff
@@ -0,0 +1,467 @@
+
+- OAuth Integration: Seamlessly authenticate with GitHub, Google, Apple, and
+  Azure/Microsoft.
+- Email & Password Support: Includes robust email+password authentication with
+  reset password functionality.
+- Easy Configuration: Set up via our configuration system.
+- Hosted UI: Use our hosted authentication UI to quickly add authentication to
+  your app.
+
+When a user signs up, we create a new object of type ``ext::auth::Identity``,
+which you can link to in your own schema. We then provide you with a token that
+can be set as the global ``ext::auth::client_token`` which will automatically
+populate another computed global called ``ext::auth::ClientTokenIdentity``
+which you can use directly in your access policies, or in your own globals.
+
+.. code-block:: sdl
+
+    using extension auth;
+
+    module default {
+        global current_customer := (
+            assert_single((
+                select Customer { * }
+                filter .identity = global ext::auth::ClientTokenIdentity
+            ))
+        );
+
+        type Customer {
+            required text: str;
+            required identity: ext::auth::Identity;
+        }
+
+        type Item {
+            required sku: str;
+            required description: str;
+        }
+
+        type Cart {
+            required customer: Customer;
+            multi items: Item {
+                quantity: int32;
+            };
+
+            access policy customer_has_full_access
+                allow all
+                using (global current_customer ?= .customer);
+        }
+    }
+
+
+Here's an example query using the TypeScript client:
+
+.. code-block:: typescript
+
+    import { createClient } from "edgedb";
+
+    declare const tokenFromAuthServer: string;
+    const client = createClient()
+      .withGlobals({
+        "ext::auth::client_token": tokenFromAuthServer
+      });
+
+    const carts = await client.query(`select Cart { * };`);
+
+
+
+pgcrypto
+--------
+
+We've added ``pgcrypto`` to our extensions. This exposes ``digest``, ``hmac``,
+``gen_salt`` and ``crypt`` functions for your hashing, ecrypting and salting
+needs.
+
+.. code-block:: edgeql-repl
+
+  edgedb> select ext::pgcrypto::digest('encrypt this', 'sha1');
+  {b'\x05\x82\xd8YLF\xe7\xd4\x12\x91\n\xdb$\xf1!v\xf9\xd4\x89\xc4'}
+  edgedb> select ext::pgcrypto::gen_salt('md5');
+  {'$1$FjNlXgX7'}
+
+Standard algorithms are "md5", "sha1", "sha224", "sha256", "sha384" and
+"sha512". Moreover, any digest algorithm OpenSSL supports is automatically
+picked up.
+
+
+pg_trgm
+-------
+
+This extension provides a bunch of functionality used to determine string
+similarity, which makes it a good text search alternative for some use cases:
+
+edgedb> with x := {'hello world', 'word hero', 'help the world'}
+....... select res := (x, ext::pg_trgm::word_similarity(x, 'hello world'))
+....... order by res.1 desc;
+{('hello world', 1), ('help the world', 0.5), ('word hero', 0.35714287)}
+
+
+
+Additional changes
```

Comment:
Would be nice to also mention the new Rust parser and more than 40% reduction in baseline server memory consumption.